lib/integrations/awsoidc/deployservice_update.go

```diff
@@ -0,0 +1,284 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awsoidc
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ecs"
+	ecsTypes "github.com/aws/aws-sdk-go-v2/service/ecs/types"
+	"github.com/aws/aws-sdk-go/aws/awsutil"
+	"github.com/gravitational/teleport/lib/modules"
+	"github.com/gravitational/trace"
+)
+
+// waitDuration specifies the amount of time to wait for a service to become healthy after an update.
+const waitDuration = time.Minute * 5
+
+// AWSRegionsList is the list of available AWS regions
+// https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.RegionsAndAvailabilityZones.html
+var AWSRegionsList = []string{
```

Comment:
The current implementation iterates over all regions to check for available Deploy Service Agents. Anyone have an idea for a more efficient way of finding all available Deploy Service Agents?

Reply:
Yes, we should iterate over all registered databases to get a list of regions.

Reply:
> To query for available ECS clusters Teleport will iterate over all databases filtering for RDS, and then extract the AWS.Region.

I was thinking this would require the same number of API calls because we'd need to query each region individually for rds databases. Is this information cached by Teleport and available for lookup?

Reply:
It should be cached, yes.

You will iterate over all Teleport Databases and for those that are `AWS`, extract its region.

This won't give you all the DatabaseService agents.
It is possible to have deployed an agent and having 0 databases.
I think we are ok with this.